### PR TITLE
[UPDATED]feat: add activity selection problem

### PR DESCRIPTION
**Pull request by chhavibansal**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: greedy_algorithms/activity_selection.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// C++ program for activity selection problem.
+
+// The following implementation assumes that the activities
+// are already sorted according to their finish time
+#include <iostream>
+using namespace std;
```
Comment (realstealthninja) [inline greedy_algorithms/activity_selection.cpp:6]:
```suggestion
```
using namespace std is bad practice and thus not allowed on this repository

File: greedy_algorithms/activity_selection.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// C++ program for activity selection problem.
+
+// The following implementation assumes that the activities
```
Comment (realstealthninja) [inline greedy_algorithms/activity_selection.cpp:3]: please add proper documentation according to the contributing.md file

File: greedy_algorithms/activity_selection.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+int main()
+{
+    int s[] = { 1, 3, 0, 5, 8, 5 };
+    int f[] = { 2, 4, 6, 7, 9, 9 };
+    int n = sizeof(s) / sizeof(s[0]);
+
+    // Function call
+    printMaxActivities(s, f, n);
+    return 0;
+}
```
Comment (realstealthninja) [inline greedy_algorithms/activity_selection.cpp:42]:
```suggestion
}

```

File: greedy_algorithms/activity_selection.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        // If this activity has start time greater than or
+        // equal to the finish time of previously selected
+        // activity, then select it
+        if (s[j] >= f[i]) {
+            cout << j << " ";
+            i = j;
+        }
+    }
+}
+
```
Comment (realstealthninja) [inline greedy_algorithms/activity_selection.cpp:31]: add tests please

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: greedy_algorithms/activity_selection.cpp (hunk 4 above)
Comment (chhavibansal) [reply greedy_algorithms/activity_selection.cpp:52]: udpated.

File: greedy_algorithms/activity_selection.cpp (hunk 2 above)
Comment (chhavibansal) [reply greedy_algorithms/activity_selection.cpp]: Updated

File: greedy_algorithms/activity_selection.cpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+{
+    //tests();
+    std::vector<int> start = {1, 3, 0, 5, 8, 5};
+    std::vector<int> finish = {2, 4, 6, 7, 9, 9};
+    // Function call
+    std::vector<int> ans = getMaxActivities(start, finish, start.size());
+    std::cout << "Following activities are selected" << "\n";
+    for(int x: ans) {
+        std::cout << x <<", ";
+    }
```
Comment (realstealthninja) [inline greedy_algorithms/activity_selection.cpp:64]:
```suggestion
    tests(); // run self test implementation 
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Please document everything according to https://www.github.com/TheAlgorithms/C-Plus-Plus/main/blob/contributing.md

File: greedy_algorithms/activity_selection.cpp (hunk 5 above)
Comment (chhavibansal) [reply greedy_algorithms/activity_selection.cpp]: @realstealthninja please review.

**Comment by chhavibansal** [comment]
@realstealthninja @Panquesito7 can u please review

File: greedy_algorithms/activity_selection.cpp [hunk 6]
```diff
@@ -0,0 +1,71 @@
+/**
+Problem Statement:
+You are given n activities with their start and finish times. Select the maximum number of activities
+that can be performed by a single person, assuming that a person can only work on a single activity at a time.
+Solution:
```
Comment (realstealthninja) [inline greedy_algorithms/activity_selection.cpp:5]: please look at this [example](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) and format your documentation accordingly

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: greedy_algorithms/activity_selection.cpp (hunk 6 above)
Comment (chhavibansal) [reply greedy_algorithms/activity_selection.cpp]: done

**Comment by chhavibansal** [comment]
@realstealthninja can u please review

**Review by realstealthninja — CHANGES_REQUESTED** [review]
This file is still not documented properly!! please check the contributing guidelines and document accordingly. thanks!

**Pull request closed without merging**